app/src/main/java/com/example/android/receitaboa/activity/NovaReceitaFotoActivity.java

```diff
@@ -141,11 +141,12 @@ public void onClick(DialogInterface dialog, int which) {
             @Override
             public void onClick(DialogInterface dialog, int which) {
 
-                //Retorna para a main activity se o usuário decidir adicionar uma foto da receita no futuro
-                Intent i = new Intent(NovaReceitaFotoActivity.this, MainActivity.class);
-                startActivity(i);
-
                 Toast.makeText(getApplicationContext(), getApplicationContext().getString(R.string.nova_receita_adicionada), Toast.LENGTH_SHORT).show();
+
+                //encerra a activity anterior ao salvar a foto da receita
+                NovaReceitaInfoActivity.addInfoReceita.finish();
+
+                finish();
```

Comment:
Código repetido, isolar em uma função única

Lembrar sempre do princípio DRY (Don't Repeat Yourse): https://dzone.com/articles/software-design-principles-dry-and-kiss